Sources/TuistCore/Utils/XCFrameworkMetadataProvider.swift

```diff
@@ -14,12 +15,14 @@ enum XCFrameworkMetadataProviderError: FatalError, Equatable {
             return "The .xcframework at path \(path.pathString) doesn't contain an Info.plist. It's possible that the .xcframework was not generated properly or that got corrupted. Please, double check with the author of the framework."
         case let .supportedArchitectureReferencesNotFound(path):
             return "Couldn't find any supported architecture references at \(path.pathString). It's possible that the .xcframework was not generated properly or that it got corrupted. Please, double check with the author of the framework."
+        case let .fileTypeNotRecognised(path):
+            return "File type at \(path.pathString) is not recognised."
```

Comment:
Nitpick: We should make the errors as obvious as possible for the users, and ideally actionable. If as a user I get this error I would not know what happened and what to do about it. For that reason, I'd suggest turning it into something along the lines of:


```
The extension of the file `xx.whatever`, which was found while parsing the xcframework `yy.xcframework`, is not supported.
```